lecture_3/template/nkt2111.tex

```diff
@@ -0,0 +1,255 @@
+%----------------------------------------
+% Write your notes here
+%----------------------------------------
+
+\section{Part 1: Guest Lecture}
+The first part of class was a guest lecture by Sidharth Sen, who works with Jake at Microsoft Research here in NYC. Sid's research focuses on algorithms and systems and networking and Jake noted that one of his more notable research topics involved weak AVL trees. Sid's background has a large data structures influence, and his lecture focused on exploring the question of how to store and manipulate data and how long things take.
+
+\vspace{3mm}
+Sid's aim for the lecture was to explore the following four topics:
+\begin{enumerate}
+    \item Computational tractability
+    \item Asymptotic order of growth
+    \item Common running times (data structures, algorithms)
+    \item Applications
+\end{enumerate}
+
+\subsection{Computational Tractability}
+
+We started by roughly defining computational tractability as how to tell how long things take. Another term associated with this idea is running time, T(n), which we take to be a function of the input size n. Sid highlighted some examples of common running times. Here, it is important to note that in general, constants do not matter. In other words, we will take $2n$ to be the same as $n$ since we are most interested in asymptotic behavior. The examples are as follows:
+
+\begin{itemize}
+    \item $2^n$
+    \begin{itemize}
+        \item exponential
+        \item not good
+        \item often happens with a brute force search or in a situation when we try all combinations of something
+    \end{itemize}
+    \item $cd^n$ where $c, d > 0$ are constants
+    \begin{itemize}
+        \item polynomial
+        \item not good when we are working with big data
+        \item theoretical computer scientists often consider this to be ok
+        \item here we do have to pay attention to the constants at least a little because it is not always obvious which algorithm is worse (ie. $20n^{100}$ versus $n^{1+0.002\logn}$)
```

Comment:
\logn will fail as a command, you need a space that it reads \log n

same problem later in the document